### ISSUE TITLE
start_function does not recognize input variables to function

### ISSUE THREAD

**Issue by akravc** [PrincetonUniversity/VST#368]
start_function throws an error, claiming that variable "1%positive" is not a local variable of the function. Error can be reproduced by checking out "commit with start_function error" from the DeepSpecDB repository, file **stringlist_instance.v**, line 220.

**Comment by bcip** [comment]
I tried to compile your code but not successful. This file you mentioned should be `verif/indices/stringlist_instance.v`. It seems there isn't  a Makefile for it. I complied files in the folder manually, including copying files from `verif/tuplekey". I tried to fix the problems but I still cannot compile `verif_stringlist.v` after all. Does your code work only for Coq 8.8 and not for Coq 8.9.1? Does your code rely on VFA library?

Despite I cannot compile your code, I looked at the issue you have. `cardinality_spec` is defined as 
```
Definition cardinality_spec 
  (ui: UnorderedIndex.index): funspec :=
  WITH sh: share, p: val, m: ui.(t)
  PRE [ 1 OF tptr ui.(t_type)]
    PROP()
    LOCAL( temp 1 p)
    SEP(ui.(t_repr) sh m p)
  POST [tint]
    PROP()
    LOCAL(temp ret_temp (Vptrofs (Ptrofs.repr (Zlength (elements(ui.(flatten) m))))))
    SEP(ui.(t_repr) sh m p).
```
So its parameter identifier is 1. And the function you want to prove is `f_stringlist_cardinality`, which is defined in `stringlist.v` and its parameter identifier is `_p`. `_p` is defined as 67, not 1. Maybe this is not allowed. I've only seen parameters defined as 1, 2 and 3 in specifications for external functions, e.g. `strlen`, `strcmp`. But the parameters in the function specifications we want to verify are defined as `_n`, `_p`, etc.
